homeassistant/components/snapcast/media_player.py

```diff
@@ -83,29 +88,29 @@ async def async_setup_entry(
     ].hass_async_add_entities = async_add_entities
 
 
-async def handle_async_join(entity, service_call):
+async def handle_async_join(entity, service_call) -> None:
     """Handle the entity service join."""
     if not isinstance(entity, SnapcastClientDevice):
         raise TypeError("Entity is not a client. Can only join clients.")
     await entity.async_join(service_call.data[ATTR_MASTER])
 
 
-async def handle_async_unjoin(entity, service_call):
+async def handle_async_unjoin(entity, service_call) -> None:
     """Handle the entity service unjoin."""
     if not isinstance(entity, SnapcastClientDevice):
         raise TypeError("Entity is not a client. Can only unjoin clients.")
     await entity.async_unjoin()
 
 
-async def handle_set_latency(entity, service_call):
+async def handle_set_latency(entity, service_call) -> None:
     """Handle the entity service set_latency."""
     if not isinstance(entity, SnapcastClientDevice):
         raise TypeError("Latency can only be set for a Snapcast client.")
     await entity.async_set_latency(service_call.data[ATTR_LATENCY])
```

Comment:
Why aren't these set as function inside of the media player object and then referenced like `snapshot` and `async_restore`?

Reply:
Good question. This was how there were originally defined and I did not change that. It was probably required in the old implementation because `SnapcastGroupDevice` didn't implement `async_set_latency` and had no base class it could fallback to.

Reply:
I moved the services into the classes but I think there's something wrong with the schema. When using developer tools to call the service the YAML generated by the UI is incorrect.

Reply:
Gotcha. Well my preference would be to leave them untouched, as I have changes ready to go that will implement grouping according to the [arch ](https://github.com/home-assistant/architecture/issues/364#issuecomment-625405413) that would make these services obsolete.